```
Error running mint automation
+ mkdir -p 11512-dc59969/gopath/src/github.com/minio
+ git -C 11512-dc59969/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 11512-dc59969/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 11512-dc59969/gopath/src/github.com/minio/minio fetch --quiet minio pull/11512/head:pr11512
+ git -C 11512-dc59969/gopath/src/github.com/minio/minio checkout --quiet pr11512
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/11512-dc59969/gopath
+ make -C 11512-dc59969/gopath/src/github.com/minio/minio --quiet
go: downloading google.golang.org/api v0.5.0
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/minio/minio-go/v7 v7.0.8
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.2
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/google/uuid v1.1.2
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/shirou/gopsutil v3.20.11+incompatible
go: downloading cloud.google.com/go v0.39.0
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/fatih/color v1.10.0
go: downloading github.com/colinmarc/hdfs/v2 v2.2.0
go: downloading github.com/minio/sio v0.2.1
go: downloading github.com/prometheus/procfs v0.2.0
go: downloading github.com/montanaflynn/stats v0.5.0
go: downloading github.com/minio/selfupdate v0.3.1
go: downloading golang.org/x/sys v0.0.0-20210119212857-b64e53b001e4
go: downloading github.com/klauspost/reedsolomon v1.9.11
go: downloading github.com/json-iterator/go v1.1.10
go: downloading github.com/nats-io/nats.go v1.10.0
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading gopkg.in/ini.v1 v1.57.0
go: downloading github.com/mattn/go-colorable v0.1.8
go: downloading github.com/streadway/amqp v1.0.0
go: downloading github.com/go-ldap/ldap v3.0.2+incompatible
go: downloading github.com/Shopify/sarama v1.27.2
go: downloading github.com/tinylib/msgp v1.1.3
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/go-ldap/ldap/v3 v3.2.4
go: downloading golang.org/x/crypto v0.0.0-20201124201722-c8d3bf9c5392
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading go.etcd.io/etcd v0.0.0-20201125193152-8a03d2e9614b
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20200615164410-66371956d46c
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading github.com/mattn/go-isatty v0.0.12
go: downloading gopkg.in/yaml.v2 v2.3.0
go: downloading google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/klauspost/cpuid v1.3.1
go: downloading github.com/golang/protobuf v1.4.3
go: downloading github.com/nats-io/nkeys v0.2.0
go: downloading github.com/minio/md5-simd v1.1.1
go: downloading github.com/klauspost/readahead v1.3.1
go: downloading github.com/nats-io/stan.go v0.7.0
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/jcmturner/gofork v1.0.0
go: downloading golang.org/x/net v0.0.0-20201216054612-986b41b23924
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/willf/bloom v2.0.3+incompatible
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: downloading github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/nsqio/go-nsq v1.0.8
go: downloading google.golang.org/protobuf v1.23.0
go: downloading google.golang.org/grpc v1.26.0
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.1
go: downloading github.com/klauspost/cpuid/v2 v2.0.3
go: downloading github.com/klauspost/compress v1.11.7
go: downloading github.com/nats-io/jwt v1.1.0
go: downloading github.com/minio/highwayhash v1.0.1
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/golang/snappy v0.0.1
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/rs/xid v1.2.1
go: downloading github.com/miekg/dns v1.1.35
go: downloading go.uber.org/zap v1.13.0
go: downloading github.com/jcmturner/goidentity/v6 v6.0.1
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/prometheus/client_golang v1.8.0
go: downloading github.com/eclipse/paho.mqtt.golang v1.3.0
go: downloading go.opencensus.io v0.22.5
go: downloading github.com/olivere/elastic/v7 v7.0.22
go: downloading github.com/hashicorp/vault/api v1.0.4
go: downloading github.com/beorn7/perks v1.0.1
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.5.0
go: downloading github.com/hashicorp/go-uuid v1.0.2
go: downloading github.com/willf/bitset v1.1.11
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading github.com/minio/simdjson-go v0.2.1
go: downloading github.com/hashicorp/vault/sdk v0.1.13
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/dswarbrick/smart v0.0.0-20190505152634-909a45200d6d
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading go.uber.org/multierr v1.3.0
go: downloading github.com/coreos/go-systemd v0.0.0-20180511133405-39ca1b05acc7
go: downloading github.com/mailru/easyjson v0.7.6
go: downloading github.com/gogo/protobuf v1.3.1
go: downloading github.com/golang/groupcache v0.0.0-20190702054246-869f871628b6
go: downloading golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0
go: downloading github.com/beevik/ntp v0.3.0
go: downloading github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/pierrec/lz4 v2.5.2+incompatible
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/prometheus/common v0.14.0
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: downloading github.com/coreos/pkg v0.0.0-20160727233714-3ac0863d7acf
go: downloading golang.org/x/time v0.0.0-20191024005414-555d28b269f0
go: downloading golang.org/x/text v0.3.3
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/gorilla/handlers v1.5.1
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading github.com/lib/pq v1.8.0
go: downloading github.com/dchest/siphash v1.2.1
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: downloading github.com/philhofer/fwd v1.1.1
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading github.com/coredns/coredns v1.4.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: downloading github.com/josharian/intern v1.0.0
go: downloading go.uber.org/atomic v1.5.0
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/coreos/go-semver v0.2.0
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading github.com/felixge/httpsnoop v1.0.1
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/gomodule/redigo v1.8.3
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: downloading github.com/ryanuber/go-glob v1.0.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/tidwall/gjson v1.6.7
go: downloading github.com/tidwall/sjson v1.0.4
go: downloading github.com/tidwall/match v1.0.3
go: downloading github.com/tidwall/pretty v1.0.2
# github.com/minio/minio/cmd
cmd/bucket-replication.go:717:30: too many arguments in call to z.getPoolIdx
	have (context.Context, string, string, ObjectOptions, int64)
	want (context.Context, string, string, int64)
make: *** [build] Error 2

```

--- a/cmd/bucket-replication.go
+++ b/cmd/bucket-replication.go
@@ -714,9 +714,7 @@ func replicateObject(ctx context.Context, objInfo ObjectInfo, objectAPI ObjectLa
 	}
 
 	// This lower level implementation is necessary to avoid write locks from CopyObject.
-	poolIdx, err := z.getPoolIdx(ctx, bucket, object, ObjectOptions{
-		VersionID: objInfo.VersionID,
-	}, objInfo.Size)
+	poolIdx, err := z.getPoolIdx(ctx, bucket, object, objInfo.Size)
 	if err != nil {
 		logger.LogIf(ctx, fmt.Errorf("Unable to update replication metadata for %s/%s(%s): %w", bucket, objInfo.Name, objInfo.VersionID, err))
 	} else {

--- a/cmd/erasure-object.go
+++ b/cmd/erasure-object.go
@@ -460,6 +460,7 @@ func (er erasureObjects) getObjectInfo(ctx context.Context, bucket, object strin
 		// Make sure to return object info to provide extra information.
 		return objInfo, toObjectErr(errMethodNotAllowed, bucket, object)
 	}
+
 	if fi.Deleted {
 		if opts.VersionID == "" || opts.DeleteMarker {
 			return objInfo, toObjectErr(errFileNotFound, bucket, object)

--- a/cmd/erasure-server-pool.go
+++ b/cmd/erasure-server-pool.go
@@ -246,31 +246,26 @@ func (z *erasureServerPools) getServerPoolsAvailableSpace(ctx context.Context, s
 
 // getPoolIdx returns the found previous object and its corresponding pool idx,
 // if none are found falls back to most available space pool.
-func (z *erasureServerPools) getPoolIdx(ctx context.Context, bucket, object string, opts ObjectOptions, size int64) (idx int, err error) {
+func (z *erasureServerPools) getPoolIdx(ctx context.Context, bucket, object string, size int64) (idx int, err error) {
 	if z.SinglePool() {
 		return 0, nil
 	}
 	for i, pool := range z.serverPools {
-		objInfo, err := pool.GetObjectInfo(ctx, bucket, object, opts)
-		switch err.(type) {
-		case VersionNotFound:
-			// VersionId not found, versionId was specified
-		case ObjectNotFound:
-			// VersionId was not specified but found delete marker or no versions exist.
-		case MethodNotAllowed:
-			// VersionId was specified but found delete marker
-		default:
-			// All other unhandled errors return right here.
-			if err != nil {
-				return -1, err
-			}
+		objInfo, err := pool.GetObjectInfo(ctx, bucket, object, ObjectOptions{})
+		if err != nil && !isErrObjectNotFound(err) {
+			return -1, err
 		}
-		// delete marker not specified means no versions
-		// exist continue to next pool.
-		if !objInfo.DeleteMarker && err != nil {
+		if isErrObjectNotFound(err) {
+			// No object exists or its a delete marker,
+			// check objInfo to confirm.
+			if objInfo.DeleteMarker && objInfo.Name != "" {
+				return i, nil
+			}
+			// objInfo is not valid, truly the object doesn't
+			// exist proceed to next pool.
 			continue
 		}
-		// Success case and when DeleteMarker is true return.
+		// object exists at this pool.
 		return i, nil
 	}
 
@@ -604,7 +599,7 @@ func (z *erasureServerPools) PutObject(ctx context.Context, bucket string, objec
 		return z.serverPools[0].PutObject(ctx, bucket, object, data, opts)
 	}
 
-	idx, err := z.getPoolIdx(ctx, bucket, object, opts, data.Size())
+	idx, err := z.getPoolIdx(ctx, bucket, object, data.Size())
 	if err != nil {
 		return ObjectInfo{}, err
 	}
@@ -625,7 +620,7 @@ func (z *erasureServerPools) DeleteObject(ctx context.Context, bucket string, ob
 	}
 
 	// We don't know the size here set 1GiB atleast.
-	idx, err := z.getPoolIdx(ctx, bucket, object, opts, 1<<30)
+	idx, err := z.getPoolIdx(ctx, bucket, object, 1<<30)
 	if err != nil {
 		return objInfo, err
 	}
@@ -676,7 +671,7 @@ func (z *erasureServerPools) CopyObject(ctx context.Context, srcBucket, srcObjec
 
 	cpSrcDstSame := isStringEqual(pathJoin(srcBucket, srcObject), pathJoin(dstBucket, dstObject))
 
-	poolIdx, err := z.getPoolIdx(ctx, dstBucket, dstObject, dstOpts, srcInfo.Size)
+	poolIdx, err := z.getPoolIdx(ctx, dstBucket, dstObject, srcInfo.Size)
 	if err != nil {
 		return objInfo, err
 	}
@@ -860,7 +855,7 @@ func (z *erasureServerPools) NewMultipartUpload(ctx context.Context, bucket, obj
 	}
 
 	// We don't know the exact size, so we ask for at least 1GiB file.
-	idx, err := z.getPoolIdx(ctx, bucket, object, opts, 1<<30)
+	idx, err := z.getPoolIdx(ctx, bucket, object, 1<<30)
 	if err != nil {
 		return "", err
 	}

